File: common/utils/ynab_api.py
from typing import Dict

import requests as requests

BASE_URL = "https://api.youneedabudget.com/v1"
BUDGETS_URL = f"{BASE_URL}/budgets/"


def create_transactions(token, budget_id, transactions):
    transactions_url = f"{BUDGETS_URL}{budget_id}/transactions"
    response = requests.post(transactions_url, headers={"Authorization": f"Bearer {token}"},
                             json={"transactions": transactions})
    response_json = response.json()
    if "error" in response_json:
        print(response_json["error"])
    return {"imported": len(response_json["data"]['transaction_ids']),
            "duplicated": len(response_json["data"]['duplicate_import_ids'])}

File: ynab_connector_app/utils/ynabApi.ts
import axios from 'axios'
const BASE_URL = "https://api.youneedabudget.com/v1"
const BUDGETS_URL = `${BASE_URL}/budgets/`

export async function getBudgets(token: string) {
    token = '85edcb6a16dbca247b161e60313367f159f7ff8ad74066d43ecfad5e4e98c759'
    const {data} = await axios.get(BUDGETS_URL, {headers: {"Authorization": `Bearer ${token}`}})
    return data.data.budgets
}

export async function getAccountsByBudgetId(token: string, budgetId: string) {
    token = '85edcb6a16dbca247b161e60313367f159f7ff8ad74066d43ecfad5e4e98c759'
    const {data} = await axios.get(`${BUDGETS_URL}/${budgetId}/accounts`, {headers: {"Authorization": `Bearer ${token}`}})
    return data.data.accounts
}